### DISCUSSION TITLE
Can I control the order of plugin wrapRootElement?

### DISCUSSION THREAD

**Discussion by aaronadamsCA** [gatsbyjs/gatsby#32196] [Help]
Scenario:

- Sentry provides an `ErrorBoundary` component.
- Theme UI provides a `ThemeProvider` component.
- I'd like `ErrorBoundary` to have access to `ThemeProvider` context, so I can use theming in `ErrorBoundary.fallback`.

I'm currently using `gatsby-plugin-theme-ui`, which uses `wrapRootElement` to add its `ThemeProvider`. As far as I can tell, this always occurs _before_ my own `wrapRootElement` call, meaning my `ErrorBoundary` _cannot_ access `ThemeProvider`'s context, since it's inside the `ErrorBoundary`.

This has also made me realize the potential for larger risks when it comes to the order in which `wrapRootElement` is executed.

I can't seem to find any docs; is the order defined, and is it something I can control?

**Comment by aaronadamsCA** [comment]
Note: I know I can just load `ThemeProvider` within the fallback, or remove the plugin and load `ThemeProvider` myself. My larger concern here is to understand `wrapRootElement` ordering, especially among multiple plugins. 👍

**Comment by herecydev** [comment]
There's no granular control over ordering, however the order of plugins in your `gatsby-config.js` determines the order of execution, so try swapping stuff around until your happy.

Of course, you could run into issues if, for another API, you required another reorder.
